ui/src/CollectionExerciseDetails.js

```diff
@@ -36,10 +37,14 @@ class CollectionExerciseDetails extends Component {
     newActionRuleType: ''
   }
 
-  componentDidMount() {
+  async componentDidMount() {
```

Comment:
I'm surprised this works... doesn't feel quite right... feels icky

Reply:
I've read through the code some more... I think you should have an async method in this class, which you call from here, instead of making the whole of `componentDidMount` async, which could cause rendering problems